dotcom-rendering/src/web/components/YoutubeBlockComponent.importable.tsx

```diff
@@ -88,11 +90,29 @@ export const YoutubeBlockComponent = ({
 	duration,
 	origin,
 	stickyVideos,
+	showOverlayCaption,
 }: Props) => {
 	const [consentState, setConsentState] = useState<ConsentState | undefined>(
 		undefined,
 	);
 
+	useEffect(
```

Comment:
If I remember correctly, we usually try to create functions outside of the useeffect so that it'll be created once and then the reference of the function is used in the useeffect. Maybe something like this: 

```
	const hideOverlayCaptionOnPlay = (showCaption: boolean | undefined) => {
		const hideOverlayCaption = () => {
			if (showCaption) {
				const caption: HTMLElement | null =
					document.querySelector('.overlay-caption');
				if (caption) caption.style.display = 'none';
			}
		};
		document.addEventListener('video:play', hideOverlayCaption);

		return () =>
			document.removeEventListener('video:play', hideOverlayCaption);
	};

	useEffect(
		() => hideOverlayCaptionOnPlay(showOverlayCaption),
		[showOverlayCaption],
	);
```

Reply:
Perfect, great tip! Thanks Marjan. I'll refactor this